django_backend/scrounger/views.py

```diff
@@ -188,6 +191,22 @@ def search_view(request):
         sim_query = boonsdk.SimilarityQuery(simhashes)
         must_queries.append(sim_query)
 
+    # Check for a query param containing uploaded assets for similarity search.
+    uploaded_assets_json = request.GET.get('uploaded_assets')
+    if uploaded_assets_json:
+        images = []
+        uploaded_assets = json.loads(unquote(uploaded_assets_json))
+        for uploaded_asset in uploaded_assets:
+            image_format, image_base64 = uploaded_asset.split(';base64,')
+            image_extension = image_format.split('/')[-1]
```

Comment:
Is this image_format a mimetype?

If so you might want to use the mimetypes lib to guess the extension type? https://docs.python.org/3/library/mimetypes.html

Reply:
the truth is, at this point, it's always gonna be a `jpeg` since the frontend has already resized and converted it
I guess I could hard code it